### PR TITLE
General refactor

### PR DESCRIPTION
**Pull request by mpivaa**
Why?

- Decouple components
- Extract components
- Higher order components(AutoWidth :heart:)
- Code re-write
- Allow user defined styles

There's still work to be done:

- [x] Define a better pattern for modules export, definition, getters/setters, props defaults
- [x] This removes multi handler ability for now
- [x] Tick, TickSet, TickCollection, TickCollectionUtils is still too complex
- [ ] svg_components is not DRY
- [x] Name convention

### CODE REVIEW

File: lib/background.jsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React from "react";
+import { Rect } from './react_svg';
+
+export default class Background extends React.Component {
+  render() {
+    return (
+      <Rect width="100%"
+        x="0" y="0"
+        height="100%"
+        {...this.props.style} />
```
Comment (jaimerson) [inline lib/background.jsx]: Oq o `...` faz?

File: lib/handler.jsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { TickCollectionUtils } from './tick_collection';
+import { Circle } from './react_svg';
+
+class Handler extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      x: null
```
Comment (jaimerson) [inline lib/handler.jsx]: `center_x` ou `position_x`

File: lib/background.jsx (hunk 1 above)
Comment (mpivaa) [inline lib/background.jsx]: `**kwargs`

**Comment by phillipp-oliveira** [comment]
sobe um exemplo no heroku ;)

File: README.md [hunk 3]
```diff
@@ -9,10 +9,34 @@ A Timeline Slider for React.
...
+```bash
+cd react-timeline-slider
+npm install
+gulp build
+```
+
+run a HTTP server
+
+```bash
+http-serve 8080
```
Comment (karydja) [inline README.md]: `server`? :smiley:

**Comment by karydja** [comment]
:shipit:

File: README.md (hunk 3 above)
Comment (mpivaa) [inline README.md]: screenshot

Comment (karydja) [inline README.md]: Vivendo e aprendendo OIHEHIOEHIEO

**Pull request merged**